packages/block-library/src/post-comments/index.php

```diff
@@ -38,6 +38,9 @@ function render_block_core_post_comments( $attributes, $content, $block ) {
 	if ( isset( $attributes['textAlign'] ) ) {
 		$classes .= 'has-text-align-' . $attributes['textAlign'];
 	}
+	if ( ! comments_open( $post_id ) && ! empty( $attributes['hideCommentsClosed'] ) && $attributes['hideCommentsClosed'] ) {
+		$classes .= 'hide-comments-closed';
+	}
```

Comment:
Would it not be more efficient to early return instead of adding an extra class? 🤔 
This way if the post doesn't have comments and we want to hide comments-closed, no HTML will be rendered (as opposed to printing it and then hide it via CSS)

```suggestion
	if ( ! comments_open( $post_id ) && ! empty( $attributes['hideCommentsClosed'] ) && $attributes['hideCommentsClosed'] ) {
		$args = array(
			'post_id' => $post_id,
			'count'   => true,
		);
		if ( ! get_comments( $args ) ) {
			return;
		}
	}
```

Reply:
I considered an early return, but decided against it because the message would still show if there are comments. 

If we're okay only hiding the message if there are no comments, then I'd much prefer this approach!

Reply:
I think that’s fine. The issue is just that the message shows when there are no comments _and_ comments are closed.

Reply:
But actually… should that just be the default behavior of the block then? Rather than something that’s opted into?

- If comments are closed, show the message. 
- If comments are closed *and* there are no comments, don’t show the message.


I can’t think a reason why you would want this message showing up under that second scenario. 🤔